src/fallbacks.jl

```diff
@@ -107,9 +107,9 @@ function coordinates(t::AbstractFeatureTrait, feature)
     geom = geometry(feature)
     isnothing(geom) ? [] : coordinates(geom)
 end
-coordinates(t::AbstractFeatureCollectionTrait, fc) = [coordinates(f) for f in getfeature(t, fc)]
+coordinates(t::AbstractFeatureCollectionTrait, fc) = map(f -> coordinates(f), getfeature(t, fc))
 
-extent(::AbstractTrait, x) = Extents.extent(x)
```

Comment:
We catch the possible `nothing` trait in the other method, so this change isn't needed?

Reply:
The idea here is that `Extents.extent` is more generic and will get the extent of a other objects like a Raster or YAXarray if they happen to be passed to `GeoInterface.extent`

Reply:
Then it relates it to https://github.com/JuliaGeo/GeoInterface.jl/pull/76? Might be good to revive it, without the whole affine discussion, just have a trait/extent for now?

Reply:
No its unrelated to #76 

This is just so that you can use any object as a bounding box anywhere that that information is available.

In Rasters.jl I have to handle this distinction in a bunch of unnessesary places, which means everyone else will have to as well.

Reply:
Can you link to that code so I can understand the issue a bit better? I don't want to hold up a pragmatic fix over pedantic architecture discussions, although I do like a clean architecture.

On a related note, I'm implementing GeoInterface for Extent.